src/tools/r2rdump/R2RMethod.cs

```diff
@@ -113,148 +129,50 @@ public R2RMethod(byte[] image, RuntimeFunction[] runtimeFunctions, ref MetadataR
             {
                 ArgTypes[i] = new SignatureType(ref signatureReader, ref mdReader);
             }
+        }
 
-            Token = MetadataTokens.GetToken(methodDefHandle);
-            int rid = MetadataTokens.GetRowNumber(methodDefHandle);
+        public R2RMethod(byte[] image, RuntimeFunction[] runtimeFunctions, NativeArray methodEntryPoints, uint offset, int rid)
+        {
+            Token = _mdtMethodDef + rid;
```

Comment:
This should be a `|` operation instead of `+` operation and I think the rid variable should be a `uint`.